File: Web/unity-web-ui/src/runtime/Session/IMessage.ts
export interface IMessage { }

export interface IRequest extends IMessage {
    requestId: number;
}

export interface IResponse extends IRequest {
    statusCode: number;
    message?: string;
}

export class MessageObject {
}

export abstract class RequestObject extends MessageObject implements IRequest {
    requestId: number = -1;
}

export class ResponseObject extends RequestObject implements IResponse {
    statusCode: number = 0;
    message?: string;

    constructor(statusCode: number = 0, message?: string) {
        super();
        this.statusCode = statusCode;
        this.message = message;
    }
}

export type MessageType<T extends MessageObject = MessageObject> = new (...args: any[]) => T;
export type RequestType<T extends RequestObject = RequestObject> = new (...args: any[]) => T;
export type ResponseType<T extends ResponseObject = ResponseObject> = new (...args: any[]) => T;

File: Server/Foundation/Base/Network/IMessage.cs
namespace Chaos;

public interface IMessage
{
}

public interface IRequest : IMessage
{
    int RequestId { get; set; }
}

public interface IResponse : IMessage
{
    int RequestId { get; set; }
    int StatusCode { get; set; }
    string Message { get; set; }
}